build.sbt

```diff
@@ -512,7 +519,19 @@ lazy val scalalib =
       scalacOptions += "-language:postfixOps",
       // The option below is needed since Scala 2.13.0.
       scalacOptions += "-language:implicitConversions",
-      scalacOptions += "-language:higherKinds"
+      scalacOptions += "-language:higherKinds",
+      /* Used to disable fatal warnings due to problems with compilation of `@nowarn` annotation */
+      scalacOptions := Def.task {
+        val prev = scalacOptions.value
+        val shouldSkipFatalWarnings = {
+          matchesScalaVersion(scalaVersion.value) {
```

Comment:
There is already `CrossVersion.partialVersion` for this in sbt. See an example here:
https://github.com/sbt/sbt/issues/6080#issuecomment-758194996